java/code/src/com/redhat/rhn/domain/errata/impl/PublishedErrata.java

```diff
@@ -57,6 +57,18 @@ public void addChannel(Channel channelIn) {
             this.channels = new HashSet();
         }
         channels.add(channelIn);
+        channelIn.getErratas().add(this);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public void removeChannel(Channel toRemove) {
+        if (this.channels == null) {
```

Comment:
This looks to me as a `dead code`, `this.channels` will never be `null` as per it is initialized to `new HashSet()`. What about a fluent logic like the thinking:
```java
if (!this.channels.isEmpty() && this.channels.contains(toRemove)) {
    channels.remove(toRemove);
    toRemove.getErratas().remove(this);
}
```

or if you really are afraid it can be `null`, simply
```java
if (this.channels != null) {
    channels.remove(toRemove);
    toRemove.getErratas().remove(this);
}
```

Reply:
I think omitting the `if` clause is enough. Fixed.